### PR TITLE
Compatibility aelia currency switcher

### PR DESCRIPTION
**Pull request by daigo75**
I've been asked by my client to open a new pull request, since the PR sent to the WooCommerce Team was closed, probably by mistake, without a merge.

### All Submissions:

* [ ] Does your code follow the [Extendables](https://extendomattic.wordpress.com/standardizations/) standards? **I think so, but can't say, the link to the standards can't be reached.**
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changes proposed in this Pull Request:
Integration with the Aelia Currency Switcher, for multi-currency support. We proposed the same integration a few months ago to Sau/Cal, but it hasn't been added to the Amazon Pay plugin, yet.

### How to test the changes in this Pull Request:
1. Install and configure the Aelia Currency Switcher (free copy available on request), enabling multiple currencies.
2. Go to WooCommerce > Settings > Payments > Amazon Pay.
3. Verify that the integration is active (see https://prnt.sc/10d2jg0). 
4. Enable Amazon Pay for some currencies in the gateway settings.
5. Enable Amazon Pay for the same currencies at WooCommerce > Currency Switcher > Payment Gateways (see https://prnt.sc/10d2izo).
6. Place an order in the enabled currencies, paying with Amazon. and verify that it's processed correctly.

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changelog entry

> Added integration with the Aelia Currency Switcher, for multi-currency support.

### CODE REVIEW

**Comment by msaggiorato** [comment]
Hello @daigo75 , can you point us to the PR that was closed by WooCommerce? Thanks

**Comment by daigo75** [comment]
@kalessil @msaggiorato Reminder that this PR is still waiting to be merged. It's ready for that, you just need to run some quick tests with your latest payment plugin. We have quite a few clients who are waiting for this, therefore I'm sending this as a formal solicitation.

**Comment by amistry007** [comment]
+100 we need this!

**Comment by kate45623** [comment]
We need this feature. Thanks!

**Comment by antweb76** [comment]
Please please please do this, it's desperately needed

**Comment by caponica** [comment]
Would be great if this could be cleaned up and merged

**Comment by RalucaSbN** [comment]
Please release this feature asap as this is needed for multiple teams.
Thanks!

**Comment by daigo75** [comment]
> Would be great if this could be cleaned up and merged

I think that the code could be merged almost as is. It worked fine when I wrote it, almost 18 months ago, now there's probably the need to address the conflict that arose due to the changes made to the files I had to update.

**Comment by AppleShane** [comment]
This feature is needed. Hoping you would make it available soon. Thanks!

**Comment by daigo75** [comment]
In reply to the person from the Amazon Pay Team, who explained their reasons for not proceeding with the integration:
1. The integration would be difficult to review and test. 
Incorrect. The integration consists of three lines of code, spread across a much larger class which is 90% documentation. 
2. he integration could require a lot of maintenance in the future, because the Currency Switcher can break it. 
Nope. The Currency Switcher is one of the few plugins that never introduced breaking changes, in over 10 years. Once the integration has been implemented, it will work indefinitely. If, and only if, significant changes are ever required, I personally inform all the developers who wrote an integration to show them how to adapt their code. 
3. The Aelia Currency Switcher is not very popular and it doesn't warrant an integration.
The Currency Switcher was the first true multi-currency plugin ever released for WooCommerce. For two whole years, between 2013 and 2015, it was also the only one, as other developers believed that multi-currency sites were going to be a fad. I'm the "pioneer" who introduced multi-currency sites, despite all the resistance, and I've been promoting the adoption of a "multi-currency mindset" for a decade. That's why the Aelia Currency Switcher not only keeps going strong, but more developers integrate with it on a regular basis. We have thousands of official installations, not to mention the "GPL Club" ones.

In conclusion, not only an integration would be easy to implement and maintain, it should actually be a priority.

**Comment by sheeSbN** [comment]
Please add this feature, it will be a great help. Thanks!

**Comment by ChristianAF** [comment]
Hey @amistry007 @kate45623 @antweb76 @sheeSbN @daigo75 @AppleShane @RalucaSbN @caponica,
Please redirect your request here: https://woocommerce.com/feature-requests/pay-with-amazon/. The decision is made by WooCommerce in addition to the Amazon team.
Thanks!

**Comment by daigo75** [comment]
@ChristianAF unfortunately, I can't use the page you indicated. No matter how I try to log in, I keep receiving an error.

If anyone else is able to file the feature request, I would like to ask them to add the following information to it:
1. A link to this conversation.
2. The information that I already wrote the integration myself 18 months ago, and it works. It's just a matter of merging it.
3. The fact that the integration will be low maintenance and there's basically no risk of it breaking due to changes in the Currency Switcher.

In addition to the above, I'm one of the most responsive developers in the WooCommerce ecosystem, answering questions seven days a week, therefore any developer who has a question just has to reach out to me. 😁

**Comment by ChristianAF** [comment]
Hey @amistry007 @kate45623 @antweb76 @sheeSbN @daigo75 @AppleShane @RalucaSbN @caponica, you can add your request here https://github.com/woocommerce/woocommerce-gateway-amazon-pay/pull/102 
Thanks

**Comment by amistry007** [comment]
I have raised the feature request on the WooCommerce site. When submitted it said the feature was awaiting approval before it will be available to vote on. WooCommerce team confirmed they work collaboratively with the developer to prioritise the features based on votes.

**Comment by daigo75** [comment]
Thanks @amistry007. Since the work has already been done, the collaboration should take no time at all. 😄

**Comment by amistry007** [comment]
@ChristianAF I have raised the feature request to WooCommerce over a week ago but still it is not showing as approved to be voted on. Can you confirm what the delay on this is?

Following is extract from woocommerce support...

> Hi there Ashok, Thanks for contacting WooCommerce.com support. I understand you submitted a feature request for Amazon Pay and this has not been approved or made public to be voted on yet. We do have a number of pending feature requests and each must be approved before being published, and we can't presently provide an ETA on when a request will be approved. If we can help with anything else, please do let us know.
